### PR TITLE
Add commit-message-formatter extension

### PR DESCRIPTION
**Pull request by zrr1999**
### Commit Message Formatter

[Commit Message Formatter](https://github.com/zrr1999/commit-message-formatter) is an commit messages generator.

#### Features

- [x] Generate commit message by LLM following selected text
- [x] Concurrency of commit message generation
- [ ] History of generated commit messages
- [ ] Support custom git types
- [ ] Integrate with git diff

#### Preferences

You can customize the following preferences:

- `Emoji Format`: Choose between `emoji` and `emoji code`.
- `Copy Format`: Choose between `emoji-type` and `emoji`.
- `Terminator`: Change the terminator of the commit type and message.
- `Action`: Choose between `Copy to Clipboard`, `Paste into Editor` and `Copy and Paste`.
- `OpenAI API Key`: API Key is used to authenticate with OpenAI API.
- `OpenAI Base Path (Optional)`: Base Path is used to override base path with OpenAI API.
- `Model`: Model is used to specify the model to use.
- `Language`: LLM will use this language to generate content.

#### Customize

You can fork this repo and change the `src/lib/types.ts` to add more gitmojis.

This PR is auto-generated by [create-pull-request][1]
## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

[1]: https://github.com/peter-evans/create-pull-request

### CODE REVIEW

**Comment by pernielsentikaer** [comment]
When I open the command I see double in the top, is that intended? I didn't select anything

commit-message-formatter 2024-10-11 at 11 54 35

Opening the other command without having finder activated crashes, could you handle that gracefully?

commit-message-formatter 2024-10-11 at 11 55 51

**Comment by zrr1999** [comment]
> When I open the command I see double in the top, is that intended? I didn't select anything
> 
> commit-message-formatter 2024-10-11 at 11 54 35
> 

It should be because I didn't filter empty inputs, and now I have fixed the issue mentioned above.

> Opening the other command without having finder activated crashes, could you handle that gracefully?
> 
> commit-message-formatter 2024-10-11 at 11 55 51

This issue should have also been fixed.

**Comment by pernielsentikaer** [comment]
Could we add this under your own account and not the org?

File: extensions/commit-message-formatter/package.json [hunk 1]
```diff
@@ -0,0 +1,172 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "commit-message-formatter",
+  "title": "Commit Message Formatter",
+  "description": "Generate and format your commit message according to the selected text with LLM",
+  "icon": "icon.png",
+  "owner": "bone-corp",
```
Comment (zrr1999) [inline extensions/commit-message-formatter/package.json:7]:
```suggestion
```

**Review by zrr1999 — COMMENTED** [review]
rm owner

**Comment by zrr1999** [comment]
> Could we add this under your own account and not the org?

ok

**Comment by pernielsentikaer** [comment]
There are still some CI errors. Can you resolve them?

**Comment by zrr1999** [comment]
> There are still some CI errors. Can you resolve them?

Alright, I have already fixed it

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**